JavaScript/Comments/app.js

```diff
@@ -0,0 +1,361 @@
+   */
+  function render(containers) {
+    return function (path, value, prevValue) {
+      switch (path) {
+        case "comments": {
+          storeData("comments", this.comments);
+        }
+        case "uiState.commentReplyId": {
+          renderCommentsList(this, containers.list);
+          renderNewCommentForm(this, containers.form);
+        }
+      }
+    };
+  }
+
+  /**
+   * this function appends comments list into DOM
+   */
+  function renderCommentsList(state, containerEl) {
+    containerEl.innerHTML = "";
+    const { comments } = state;
+
+    const rootComments = comments.filter((comment) => !comment.asReplyFor);
+
+    const commentsWithReplies = rootComments
+      .map((comment) => {
+        const replies = comments
+          .filter((c) => c.asReplyFor === comment.id)
+          .sort((a, b) => b.createdAt - a.createdAt);
+        return { ...comment, replies };
+      })
+      .sort((a, b) => b.createdAt - a.createdAt);
+
+    const commentsListEl = createCommentsList(state, commentsWithReplies);
+
+    containerEl.append(commentsListEl);
+  }
+
+  /**
+   * this function creates, configure and returns comments list
+   * but NOT appends it to DOM
+   */
+  function createCommentsList(state, comments) {
+    const listEl = createElement("ul");
+
+    const commentsCollectionEl = comments.map((comment) => {
+      const listItemEl = createElement("li");
+      const commentEl = createSingleComment(state, comment);
+      const hasReplies = comment.replies ?? false;
+
+      listItemEl.append(commentEl);
+
+      if (hasReplies && hasReplies.length !== 0) {
+        const replies = createCommentsList(state, comment.replies);
+        listItemEl.append(replies);
+      }
+
+      return listItemEl;
+    });
+
+    listEl.append(...commentsCollectionEl);
+    return listEl;
+  }
+
+  /**
+   * this function creates, configure and returns comment card
+   * but NOT appends it to DOM
+   */
+  function createSingleComment(state, commentData) {
+    const elementsConfig = [
+      {
+        tagName: "article",
+        options: {
+          className: commentData.asReplyFor
+            ? "comment-card comments__card comment-card--reply"
+            : "comment-card comments__card",
+        },
+      },
+      {
+        tagName: "h3",
+        options: {
+          className: "comment-card__author",
+          textContent: commentData.author,
+        },
+      },
+      {
+        tagName: "time",
+        options: {
+          className: "comment-card__publish-date",
+          textContent: new Date(commentData.createdAt).toLocaleString(),
+        },
+      },
+      {
+        tagName: "div",
+        options: {
+          className: "comment-card__text",
+          textContent: commentData.comment,
+        },
+      },
+      {
+        tagName: "button",
+        options: {
+          className: "btn btn--yellow btn--small comment-card__btn",
+          textContent: "Add reply",
```

Comment:
Buttons like "Add reply" and "Delete" should have aria-label attributes to be more accessible to users with screen readers